### PR TITLE
HiveDataNode node addition to support adl and abfs URI for gobblin-se…

### PR DESCRIPTION
**Pull request by pawanbtej**
…rvice

Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1003

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):
   HiveDataNode currently extends HdfsDatanode which supports only HDFS URI, this PR is to allow HiveDataNode 

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
  Test cases are built to test uri validation to true on correct URI and validate for exceptions for invalid URI

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java [hunk 1]
```diff
@@ -17,30 +17,33 @@
...
 import org.apache.gobblin.annotation.Alpha;
 import org.apache.gobblin.service.modules.flowgraph.FlowGraphConfigurationKeys;
-import org.apache.gobblin.service.modules.flowgraph.datanodes.fs.HdfsDataNode;
+import org.apache.gobblin.service.modules.flowgraph.datanodes.fs.FileSystemDataNode;
 import org.apache.gobblin.util.ConfigUtils;
 
 
 /**
- * An abstract {@link HiveDataNode} implementation. In addition to the required properties of a {@link HdfsDataNode}, an {@link HiveDataNode}
+ * An abstract {@link HiveDataNode} implementation. In addition to the required properties of a {@link FileSystemDataNode}, an {@link HiveDataNode}
```
Comment (sv2000) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java]: Drop "abstract" from the comment. It is a little misleading.

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java [hunk 2]
```diff
@@ -17,30 +17,33 @@
...
  * must have a metastore URI specified.
  */
 @Alpha
 @EqualsAndHashCode (callSuper = true)
-public class HiveDataNode extends HdfsDataNode {
+public class HiveDataNode extends FileSystemDataNode {
```
Comment (sv2000) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java]: Should HiveDataNode just extend BaseDataNode?

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java [hunk 3]
```diff
@@ -17,30 +17,33 @@
...
   public static final String METASTORE_URI_KEY = FlowGraphConfigurationKeys.DATA_NODE_PREFIX + "hive.metastore.uri";
+  private static final String[] HIVE_SUPPORTED_SCHEME = {"adl", "abfs", "hdfs"};
```
Comment (sv2000) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java]: Maybe we should drop supported schemes check. There could be other schemes (e.g. Hive on AWS) which the current list misses.

Comment (pawanbtej) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java]: removed the check and related test cases

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java (hunk 2 above)
Comment (pawanbtej) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java]: this has been fixed to extend BaseDataNode

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java (hunk 1 above)
Comment (pawanbtej) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java]: fixed

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java [hunk 4]
```diff
@@ -24,24 +24,23 @@
...
- * An abstract {@link HiveDataNode} implementation. In addition to the required properties of a {@link FileSystemDataNode}, an {@link HiveDataNode}
+ * An {@link HiveDataNode} implementation. In addition to the required properties of a {@link BaseDataNode}, an {@link HiveDataNode}
  * must have a metastore URI specified.
  */
 @Alpha
 @EqualsAndHashCode (callSuper = true)
-public class HiveDataNode extends FileSystemDataNode {
+public class HiveDataNode extends BaseDataNode {
   public static final String METASTORE_URI_KEY = FlowGraphConfigurationKeys.DATA_NODE_PREFIX + "hive.metastore.uri";
   private static final String[] HIVE_SUPPORTED_SCHEME = {"adl", "abfs", "hdfs"};
```
Comment (sv2000) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java]: Is HIVE_SUPPORTED_SCHEME still needed?

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java [hunk 5]
```diff
@@ -82,23 +81,4 @@ public boolean isMetastoreUriValid(URI metastoreUri) {
...
-
-  @Override
-  public boolean isUriValid(URI fsUri) {
-    String scheme = fsUri.getScheme();
-    //Check that the scheme is "adl"
-    if (!Arrays.asList(HIVE_SUPPORTED_SCHEME).contains(scheme)) {
-      return false;
-    }
-    //Ensure that the authority is not empty
-    if (com.google.common.base.Strings.isNullOrEmpty(fsUri.getAuthority())) {
```
Comment (sv2000) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java]: Maybe useful to keep the authority check?

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java [hunk 6]
```diff
@@ -28,19 +29,20 @@
...
- * An abstract {@link HiveDataNode} implementation. In addition to the required properties of a {@link HdfsDataNode}, an {@link HiveDataNode}
+ * An {@link HiveDataNode} implementation. In addition to the required properties of a {@link BaseDataNode}, an {@link HiveDataNode}
  * must have a metastore URI specified.
  */
 @Alpha
 @EqualsAndHashCode (callSuper = true)
-public class HiveDataNode extends HdfsDataNode {
+public class HiveDataNode extends BaseDataNode {
   public static final String METASTORE_URI_KEY = FlowGraphConfigurationKeys.DATA_NODE_PREFIX + "hive.metastore.uri";
+  private static final String[] HIVE_SUPPORTED_SCHEME = {"adl", "abfs", "hdfs"};
```
Comment (sv2000) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java]: Is this line still needed?

**Pull request closed without merging**